File: src/main/java/me/glorantq/aboe/chat/client/commands/emotes/EmoteCommand.java
package me.glorantq.aboe.chat.client.commands.emotes;

import me.glorantq.aboe.chat.client.commands.ABOEClientCommand;
import net.minecraft.client.Minecraft;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.ChatComponentText;

import java.util.ArrayList;
import java.util.List;

public class EmoteCommand extends ABOEClientCommand {
    private final String emoteName;
    private final String emote;

    public EmoteCommand(String emoteName, String emote) {
        this.emoteName = emoteName;
        this.emote = emote;
    }

    @Override
    public String getCommandName() {
        return emoteName;
    }

    @Override
    public String getCommandUsage(ICommandSender p_71518_1_) {
        return "/" + emoteName;
    }

    @Override
    public List getCommandAliases() {
        return new ArrayList();
    }

    @Override
    public void processCommand(ICommandSender p_71515_1_, String[] p_71515_2_) {
        if(!(p_71515_1_ instanceof EntityPlayer)) {
            p_71515_1_.addChatMessage(new ChatComponentText("Only players can use this command!"));
            return;
        }

        Minecraft.getMinecraft().thePlayer.sendChatMessage(emote);
    }

    @Override
    public List addTabCompletionOptions(ICommandSender p_71516_1_, String[] p_71516_2_) {
        return new ArrayList();
    }
}

File: src/main/kotlin/me/glorantq/aboe/chat/client/commands/emotes/EmoteCommand.kt
package me.glorantq.aboe.chat.client.commands.emotes

import me.glorantq.aboe.chat.client.commands.ABOEClientCommand
import net.minecraft.client.Minecraft
import net.minecraft.command.ICommandSender
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.ChatComponentText

class EmoteCommand(private val emoteName: String, private val emote: String) : ABOEClientCommand() {
    override fun getCommandName(): String = emoteName
    override fun getCommandUsage(p_71518_1_: ICommandSender): String = "/$emoteName"
    override fun getCommandAliases(): List<*> = arrayListOf<Any>()
    override fun addTabCompletionOptions(p_71516_1_: ICommandSender, p_71516_2_: Array<String>): List<*> = arrayListOf<Any>()

    override fun processCommand(p_71515_1_: ICommandSender, p_71515_2_: Array<String>) {
        if (p_71515_1_ !is EntityPlayer) {
            p_71515_1_.addChatMessage(ChatComponentText("Only players can use this command!"))
            return
        }

        Minecraft.getMinecraft().thePlayer.sendChatMessage(emote)
    }
}
